yt_dlp/extractor/gamedevtv.py

```diff
@@ -0,0 +1,108 @@
+import json
+
+from .common import InfoExtractor
+from ..networking.exceptions import HTTPError
+from ..utils import (
+    ExtractorError,
+    clean_html,
+    int_or_none,
+    join_nonempty,
+    parse_iso8601,
+    str_or_none,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class GameDevTVDashboardIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gamedev\.tv/dashboard/courses/(?P<id>\d+)'
```

Comment:
Imo. the extractor should support individual videos as well.
Given that only the course api request is needed, I think a single extractor is fine. Just add an optional sub_id to the regex and, if present, only return a single video.

Reply:
I am very open as to how that might be achieved but there really isnt a way for the user to know the IDs of a single video. You'd have to open up the dev tools in the web browser and examin the site json in order to determine that information

The UI is completely dynamic and generated on the fly in javascript. The URL does not change every. From the users' perspective, the player loads a playlist and you click the playlist without the web page ever reloading

Reply:
For me the URL does change:
`https://www.gamedev.tv/dashboard/courses/63` is the main course URL
and then clicking on the different lectures, the URL in the browser does change between:
```
https://www.gamedev.tv/dashboard/courses/63/2279
https://www.gamedev.tv/dashboard/courses/63/2280
https://www.gamedev.tv/dashboard/courses/63/2281
https://www.gamedev.tv/dashboard/courses/63/2282
https://www.gamedev.tv/dashboard/courses/63/2283
```
Those are the IDs of the different lectures. We can still just fetch the entire course (afaik. there is no API for the individual lectures) and then, if the lecture_id is passed, filter the playlist to only return a single result with the matching id.

Reply:
Thats interesting, the entire time i was working on it, the website did not exhibit this behaviour.

I'll see what I can come up with to accommodate this request

Reply:
The regex probably looks like this `r"https?://(?:www\.)?gamedev\.tv/dashboard/courses/(?P<id>\d+)/?(?P<sub_id>\d*)"`

I am struggling to see how I would implement your suggestion however. _match_id() will get the course id, but I didn't see anything that would help in this case. 

I was considering filtering the data dict and removing everything before the object is traversed but i suspect I am missing some built in facility that yt-dlp has